### PR TITLE
Add tests for TRSettingsController

### PR DESCRIPTION
**Pull request by sharplet**
This is the first in a series of PRs to facilitate porting more files to Swift. A future PR will actually port TRSettingsController to Swift; this PR is intended to add test coverage.

While adding tests I also ran into some issues with build settings related to the PCH file. This PR also removes PCH files in favour of explicit imports.

### CODE REVIEW

File: Tropos/Controllers/TRSettingsController.m [hunk 1]
```diff
@@ -2,38 +2,32 @@
...
-#pragma mark - Properties
-
-- (RACSignal *)unitSystemChanged
+- (instancetype)init
 {
-    return [[[self.userDefaultsChanged filter:^BOOL(NSNotification *notification) {
-        return [notification.name isEqualToString:NSUserDefaultsDidChangeNotification];
-    }] map:^id(id value) {
-        return @([self unitSystem]);
-    }] startWith:@([self unitSystem])];
```
Comment (sharplet) [inline Tropos/Controllers/TRSettingsController.m]: This method has moved [here](https://github.com/thoughtbot/Tropos/pull/167/files#diff-50402482dc3839f4d529b6e1a18d494eR7).

File: Tropos/Controllers/TRAppDelegate.m [hunk 2]
```diff
@@ -44,4 +49,10 @@ - (void)application:(UIApplication *)application performFetchWithCompletionHandl
     }];
 }
 
+- (BOOL)isCurrentlyTesting
+{
+    NSString *bundleInjectionPath = [[[NSProcessInfo processInfo] environment] objectForKey:@"XCInjectBundleInto"];
```
Comment (dazmuda) [inline Tropos/Controllers/TRAppDelegate.m]: Could you maybe use a run argument or preproc macro instead?

File: Tropos/Controllers/TRWeatherController.m [hunk 3]
```diff
@@ -37,6 +39,7 @@ - (instancetype)init
     self.geocodeController = [TRGeocodeController new];
     self.forecastController = [TRForecastController new];
     self.settingsController = [TRSettingsController new];
+    self.unitSystemChanged = [[self.settingsController unitSystemChanged] replayLastLazily];
```
Comment (dazmuda) [inline Tropos/Controllers/TRWeatherController.m]: Does this need to be lazy? As in, is there a reason we wait for a subscription to open the multicast connection or is just for cleanliness?

**Comment by dazmuda** [comment]
Small questions. Looks good.

File: Tropos/Controllers/TRAppDelegate.m (hunk 2 above)
Comment (sharplet) [inline Tropos/Controllers/TRAppDelegate.m]: Good idea, check out 832f613

File: Tropos/Controllers/TRWeatherController.m (hunk 3 above)
Comment (sharplet) [inline Tropos/Controllers/TRWeatherController.m]: It certainly wouldn't break without it.

Mainly: I'm not aware of any downside to the laziness, and I think deferring the work until it's actually needed is generally a good idea.

**Comment by sharplet** [comment]
Thanks @dazmuda ✨

**Pull request closed without merging**